semeio/workflows/ahm_analysis/ahmanalysis.py

```diff
@@ -0,0 +1,456 @@
+            )
+
+            # Calculate Ks matrix for scalar parameters
+            ks_data = calc_ks(
+                ks_data,
+                dkeysf,
+                prior_data,
+                GenKwCollector.loadAllGenKwData(ert, target_name),
+                group_name,
+            )
+            field_output[group_name] = _get_field_params(
+                ert, facade.get_ensemble_size(), field_parameters, target_name
+            )
+
+        # Calculate Ks matrix for Fields parameters
+        if field_parameters:
+            # Get grid characteristics to be able to plot field avg maps
+            mygrid_ok = get_field_grid_char(ert.eclConfig().get_gridfile())
+            all_input_prior = _get_field_params(
+                ert, facade.get_ensemble_size(), field_parameters, prior_name
+            )
+
+            for fieldparam in field_parameters:
+                scaler = StandardScaler()
+                scaler.fit(all_input_prior[fieldparam])
+                pca = PCA(0.98).fit(
+                    pd.DataFrame(scaler.transform(all_input_prior[fieldparam]))
+                )
+                pc_fieldprior_df = pd.DataFrame(
+                    data=pca.transform(scaler.transform(all_input_prior[fieldparam]))
+                )
+                all_ks = pd.DataFrame()
+                # Get the posterior Field parameters
+                mygrid_ok_short = mygrid_ok[mygrid_ok["KZ"] == 1].copy().reset_index()
+                for group_name in combinations.keys():
+
+                    mygrid_ok_short = calc_delta_grid(
+                        field_output[group_name][fieldparam],
+                        all_input_prior[fieldparam],
+                        mygrid_ok,
+                        group_name,
+                        mygrid_ok_short,
+                    )
+                    pc_fieldpost_df = pd.DataFrame(
+                        data=pca.transform(
+                            scaler.transform(field_output[group_name][fieldparam])
+                        )
+                    )
+                    all_ks = calc_ks(
+                        all_ks,
+                        pc_fieldpost_df,
+                        pc_fieldprior_df,
+                        pc_fieldpost_df,
+                        group_name,
+                    )
+                # add the field max Ks to the scalar Ks matrix
+                ks_data.loc["FIELD_" + fieldparam] = all_ks.max()
+                self.reporter.publish_csv("delta_field" + fieldparam, mygrid_ok_short)
+        # save/export the Ks matrix, active_obs, misfitval and prior data
+        self.reporter.publish_csv("ks", ks_data)
+        self.reporter.publish_csv("active_obs_info", active_obs)
+        self.reporter.publish_csv("misfit_obs_info", misfitval)
+        self.reporter.publish_csv("prior", prior_data)
+
+
+def _run_ministep(
+    ert, facade, obs_group, data_parameters, prior_name, target_name, output_path
```

Comment:
I don't think facade is necessary here.. Can't we just call `ert` directly? If we were able to completely swap out `ert` with facade I would agree using that instead, but otherwise I think using `ert.getLocalConfig().getUpdatestep().attachMinistep(ministep)`

And considering the `getLocalConfig`, let's make a `local_config = ert.getLocalConfig()` while we're at it and use that instead :)

Reply:
done